### DISCUSSION TITLE
Ignore components during build

### DISCUSSION THREAD

**Discussion by rinu** [nuxt/nuxt#8042] [General]
Is it possible to ignore some components during build time to make the build faster?
My project can build and run in serveral configurations for slightly different websites. The components for each are organized in ~/components into their own folders.
The ignore configuration seems to specifially exclude components. Messing with webpack config also didn't work because it only looks at the file extension.

**Comment by rinu** [comment]
To answer my own question, it seems like moving components outside of the `components` directory and using the `components:dirs` hook is the answer.

In `nuxt.config.js`
```
    components: true,
    hooks: {
      'components:dirs': (dirs) => {
        dirs.push(`~/themes/${THEME}/components`)
      }
    }
```

**Reply by rinu** [reply]
Actually just moving components to outside of the `components` directory did it. Messing with nuxt config doesn't make a difference in build or website speed. Nuxt build is a mystery.

**Reply by LeeGrobler** [reply]
Wait, when you say "moving components to outside of the components directory", d'ya mean that all your .vue files are now just floating around in your project root, or did you just rename the `components` directory to something else?

**Reply by rinu** [reply]
It used to be `components/theme/components` and became `themes/theme/components`. The other `components` at the end might seem redundant but there's also a `layouts` folder because my layouts are standard vue components rather than using nuxt's built-in layouts feature.

I now also use the standard `components` directory for shared components that are not theme specific. 

My component loader has other complexities beyond this. All to please SEO people with a better lighthouse score.

**Comment by kiddten** [comment]
any other ideas?
